backend/src/main/scala/com/softwaremill/adopttapir/infrastructure/Json.scala

```diff
@@ -1,12 +1,23 @@
 package com.softwaremill.adopttapir.infrastructure
 
 import com.softwaremill.tagging.@@
-import io.circe.generic.AutoDerivation
 import io.circe.{Decoder, Encoder, Printer}
+import io.circe.generic.extras.{AutoDerivation, Configuration => CirceConfiguration}
+import sttp.tapir.generic.{Configuration => TapirConfiguration}
 
 /** Import the members of this object when doing JSON serialisation or deserialisation.
   */
 object Json extends AutoDerivation {
+
+  implicit lazy val circeConfiguration: CirceConfiguration =
+    CirceConfiguration.default.withDiscriminator("type")
+      .copy(transformConstructorNames = (s => s.toLowerCase)
+    )
+
+  implicit val tapirConfiguration: TapirConfiguration =
```

Comment:
shouldn't this go to `http`? It doesn't have to do anything with json serialisation

Reply:
it should, moved it there.